### PR TITLE
Modernize CalculatePlaczekSelfScattering

### PR DESCRIPTION
**Pull request by peterfpeterson**
Refactor code to use more of the `Kernel::Material` class and fix a bug in calculate the summation term. This also added the function `Kernel::Material::totalAtoms()` as it is a reoccurring theme in many calculations involving material.

**To test:**

The tests should still pass as this is just a refactor of existing code.

*There is no associated issue.*

The version into `ornl-next` is #32225.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp [hunk 1]
```diff
@@ -75,35 +74,40 @@ std::map<std::string, std::string> CalculatePlaczekSelfScattering::validateInput
...
-  double densityRatio = 1.0;
-  if (crystalDensity > 0) {
-    densityRatio = numberDensity / crystalDensity;
-  }
   // get incident spectrum and 1st derivative
   const MantidVec xLambda = incidentWS->readX(0);
   const MantidVec incident = incidentWS->readY(0);
   const MantidVec incidentPrime = incidentWS->readY(1);
-  double dx = (xLambda[1] - xLambda[0]) / 2.0;
+  const double dx = (xLambda[1] - xLambda[0]) / 2.0; // assume constant bin width
```
Comment (KedoKudo) [inline Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp:110]: Just a comment here, if someone pass in a workspace with a spectrum of 1 bin, we will have segmentation fault here.

File: Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp [hunk 2]
```diff
@@ -112,7 +116,7 @@ void CalculatePlaczekSelfScattering::exec() {
   std::vector<double> eps1;
   constexpr auto LambdaD = 1.44;
```
Comment (KedoKudo) [inline Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp:117]: should we move this constexpr into the anonymous namespace above?  Also, I am kind of curious where does this 1.44 comes from :thinking:

**Review by KedoKudo — APPROVED** [review]
Changes are straight forward.  Only two comments about minor details, which do not need to be addressed at this moment.

File: Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp (hunk 1 above)
Comment (peterfpeterson) [reply Framework/Algorithms/src/CalculatePlaczekSelfScattering.cpp:110]: Ouch!

**Pull request merged**